amino.go

```diff
@@ -524,22 +524,22 @@ func (cdc *Codec) UnmarshalJSON(bz []byte, ptr interface{}) error {
 	}
 	rv = rv.Elem()
 	rt := rv.Type()
-	info, err := cdc.getTypeInfo_wlock(rt)
+	info, err := cdc.getTypeInfoWlock(rt)
 	if err != nil {
 		return err
 	}
 	// If registered concrete, consume and verify type wrapper.
 	if info.Registered {
 		// Consume type wrapper info.
-		name, bz_, err := decodeInterfaceJSON(bz)
+		name, data, err := decodeInterfaceJSON(bz)
 		if err != nil {
 			return err
 		}
 		// Check name against info.
 		if name != info.Name {
-			return fmt.Errorf("UnmarshalJSON wants to decode a %v but found a %v", info.Name, name)
+			return errors.Errorf("unmarshalJSON wants to decode a %v but found a %v", info.Name, name)
```

Comment:
```suggestion
			return errors.Errorf("UnmarshalJSON wants to decode %v, but found %v", info.Name, name)
```

Reply:
errors can not begin with a capital letter